tests/v2/test_02_models/test_model_job.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from fractal_server.app.models.v2 import DatasetV2
+from fractal_server.app.models.v2 import JobV2
+from fractal_server.app.models.v2 import ProjectV2
+from fractal_server.app.models.v2 import WorkflowV2
+from fractal_server.app.schemas.v2.job import JobStatusType
+
+
+async def test_unique_job_submitted_per_dataset(db, local_resource_profile_db):
+    resource, _ = local_resource_profile_db
+
+    project = ProjectV2(name="Project", resource_id=resource.id)
+    db.add(project)
+    await db.commit()
+    await db.refresh(project)
+
+    workflow = WorkflowV2(name="Workflow", project_id=project.id)
+    dataset1 = DatasetV2(
+        name="Dataset1", project_id=project.id, zarr_dir="/fake"
+    )
+    dataset2 = DatasetV2(
+        name="Dataset2", project_id=project.id, zarr_dir="/fake"
+    )
+    db.add_all([workflow, dataset1, dataset2])
+    await db.commit()
+    await db.refresh(workflow)
+    await db.refresh(dataset1)
+    await db.refresh(dataset2)
+
+    dataset1_id = dataset1.id
+    dataset2_id = dataset2.id
+
+    common_args = dict(
+        project_id=project.id,
+        workflow_id=workflow.id,
+        user_email="user@example.org",
+        dataset_dump={},
+        workflow_dump={},
+        project_dump={},
+        first_task_index=0,
+        last_task_index=0,
+        attribute_filters={},
+        type_filters={},
+    )
+
```

Comment:
Let's add another case:
```
job1: dataset_id = null, status = submitted
job2: dataset_id = null, status = submitted
```
what is the outcome?

(to be clear: this situation should never take place, but it's best to be aware of what it would lead to)